### PR TITLE
LPF-540:  Integration tests for /csv/{id} endpoint

### PR DESCRIPTION
**Pull request by SineadCummings**
Add integration tests for /csv/{id} endpoint

### CODE REVIEW

File: src/test/java/uk/gov/laa/gpfd/integration/OAuth2TestConfig.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Bean
+    public ClientRegistrationRepository clientRegistrationRepository() {
+        ClientRegistration graphRegistration = ClientRegistration
+                .withRegistrationId("graph")
+                .clientId("test-client-id")
+                .authorizationUri("authorizationUri")
+                .tokenUri("tokenUri")
+                .redirectUri("redirectUri")
+                .authorizationGrantType(AuthorizationGrantType.AUTHORIZATION_CODE) // Specify the authorization grant type
```
Comment (ChrisCookOC) [inline src/test/java/uk/gov/laa/gpfd/integration/OAuth2TestConfig.java:29]: Might be an unnecessary comment?

File: src/test/java/uk/gov/laa/gpfd/integration/ServerSideErrorIT.java [hunk 2]
```diff
@@ -37,4 +55,44 @@ void getReportWithIdShouldReturn500WhenCannotConnectToDb() throws Exception {
 
         Assertions.assertEquals(500, response.getStatus());
     }
+
+    @Test
```
Comment (ChrisCookOC) [inline src/test/java/uk/gov/laa/gpfd/integration/ServerSideErrorIT.java:59]: Thought - if these tests were run in a different order, would this test break? 
So if test below it created a db then this one ran, would it be upset?

Comment (SineadCummings) [reply src/test/java/uk/gov/laa/gpfd/integration/ServerSideErrorIT.java:59]: For that to happen we would need to have set up parallel execution. Whilst the tests may be executed in a different order getCsvWithIdShouldReturn500WhenCannotConnectToDbForReportTable both sets up and drops the tables required for testing. As a result, no matter when getCsvWithIdShouldReturn500WhenCannotConnectToDbForMappingTable runs, there will not be a mapping table to point at

Comment (ChrisCookOC) [reply src/test/java/uk/gov/laa/gpfd/integration/ServerSideErrorIT.java:59]: oh yeah it drops the tables thats fine :)

**Review by ChrisCookOC — APPROVED** [review]

**Pull request merged**